src/dc_receive_imf.rs

```diff
@@ -1129,8 +1131,8 @@ async fn create_or_lookup_group(
         set_better_msg(mime_parser, &better_msg);
     }
 
-    let grpid = if let Some(grpid) = mime_parser.get(HeaderDef::ChatGroupId) {
-        grpid.clone()
+    let grpid = if let Some(grpid) = try_getting_grpid(mime_parser) {
+        grpid
```

Comment:
my first idea was to move the "last resort" check below the get_parent_message check, however, at the end it probably does not really matter :)